test/core/worker.js

```diff
@@ -113,6 +113,45 @@ describe('worker', function(){
 
     });
 
+
+    describe('integration with multiple failed queues', function(){
+
+      beforeEach(function(done){
+          worker = new specHelper.NR.worker({connection: specHelper.connectionDetails, timeout: specHelper.timeout, queues: specHelper.queue, multipleFailureQueues: true}, jobs, function(){
+          done();
+        });
+      });
+
+      afterEach(function(done){
+        worker.end(function(){
+          done();
+        });
+      });
+
+      it('will not work jobs that are not defined', function(done){
+        var listener = worker.on('failure', function(q, job, failure){
+          q.should.equal(specHelper.queue);
+          String(failure).should.equal == "Error: No job defined for class 'somethingFake'";
```

Comment:
It probably should, however, the test that I ripped off had this shape :disappointed: